### PR TITLE
feat: Macro per type

### LINKED ISSUE #37: Add option to make macros post, comment or modmail specific.
Probably just add checkbox for each instance where if none are checked it will assume it needs to show on all. 
Can probably be added without a schema change if we add it like that.

### PR DESCRIPTION
**Pull request by SpyTec**
Closes #37

Adds context options for creating and editing macros. By default it will all be false for both adding and editing.
image

Existing macros which have not been edited to have these new fields will show on everything. 
image

### CODE REVIEW

**Comment by creesch** [comment]
Don't see any obvious issues with this. I'll do some actual testing later today and then merge.

**Comment by Geo1088** [comment]
Conflict caused by merging #297, resolving...

**Comment by Geo1088** [comment]
Hmm, actually, I take back my approval. I'm having trouble getting a macro with "Modmail" context to show up in the dropdown in modmail threads.

File: extension/data/modules/macros.js [hunk 1]
```diff
@@ -163,7 +178,7 @@ function modmacros () {
                     const macroButtonHtml = `<select class="tb-macro-select tb-action-button" data-subreddit="${info.subreddit}"><option value=${MACROS}>macros</option></select>`;
                     $body.find('.ThreadViewerReplyForm__replyOptions').after(`<div class="tb-usertext-buttons tb-macro-newmm">${macroButtonHtml}</div>`);
 
-                    populateSelect('.tb-macro-select', info.subreddit, config);
+                    populateSelect('.tb-macro-select', info.subreddit, config, 'comment');
```
Comment (Geo1088) [inline extension/data/modules/macros.js:181]: Unless I'm misunderstanding something, this line requires macros to have the "comment" type to work in new modmail, which seems wrong. Changing it to "modmail" fixes it in the case I was testing, at least. Would like @SpyTec to check my logic here though.

```suggestion
                    populateSelect('.tb-macro-select', info.subreddit, config, 'modmail');
```

Comment (SpyTec) [reply extension/data/modules/macros.js]: Good catch, double checked and you're right

**Pull request merged**